smslibrary/src/test/java/com/eis/smslibrary/SMSPeerTest.java

```diff
@@ -1,142 +1,128 @@
 
-    @Test
-    public void checkPhoneNumber_telephoneNumber_isEmpty() {
-        Assert.assertEquals(SMSPeer.TelephoneNumberState.TELEPHONE_NUMBER_TOO_SHORT, SMSPeer.checkPhoneNumber(EMPTY_TELEPHONE_NUMBER));
-    }
+    private static PhoneNumberUtil phoneNumberUtil = PhoneNumberUtil.getInstance();
+    private static String[] randomValidAddresses = new String[VALID_TEST_ITERATIONS];
+    private static String randomInvalidAddress;
 
-    @Test
-    public void staticCheckPhoneNumber_telephoneNumber_isValid() {
-        Assert.assertEquals(SMSPeer.TelephoneNumberState.TELEPHONE_NUMBER_VALID, SMSPeer.checkPhoneNumber(VALID_TELEPHONE_NUMBER));
+    @BeforeClass
+    public static void setup() {
+        Set<String> availableCountries = phoneNumberUtil.getSupportedRegions();
+        String[] countryCodes = availableCountries.toArray(new String[0]);
+        int randomIndex = new Random().nextInt(countryCodes.length);
+
+        randomInvalidAddress = phoneNumberUtil.format(
+                phoneNumberUtil.getInvalidExampleNumber(countryCodes[randomIndex]),
+                PhoneNumberUtil.PhoneNumberFormat.E164);
+
+        for (int i = 0; i < VALID_TEST_ITERATIONS; i++) {
+            randomIndex = new Random().nextInt(countryCodes.length);
+            randomValidAddresses[i] = phoneNumberUtil.format(
+                    phoneNumberUtil.getExampleNumber(countryCodes[randomIndex]),
+                    PhoneNumberUtil.PhoneNumberFormat.E164);
+        }
     }
 
     @Test
-    public void constructor_telephoneNumber_noException() {
-        new SMSPeer(VALID_TELEPHONE_NUMBER);
+    public void validPeers_areCreated() {
+        // Random cases
+        for(String validAddress : randomValidAddresses)
+            assertEquals(validAddress, new SMSPeer(validAddress).getAddress());
+        // Special cases
+        assertEquals(VALID_ADDR_WITH_PREFIX, new SMSPeer(VALID_ADDR).getAddress());
+        assertEquals(VALID_ADDR_WITH_PREFIX, new SMSPeer(VALID_ADDR_WITH_PREFIX).getAddress());
+        assertEquals(VALID_ADDR_WITH_PREFIX, new SMSPeer(VALID_ADDR_WITH_PREFIX_AND_SPACES).getAddress());
+        assertEquals(VALID_ADDR_WITH_PREFIX, new SMSPeer(VALID_ADDR_WITH_DASHES).getAddress());
+        assertEquals(VALID_ADDR_WITH_PREFIX, new SMSPeer(VALID_ADDR_WITH_PREFIX_AND_DASHES).getAddress());
+        assertEquals(EMULATOR_ADDR_WITH_PREFIX_AND_EXT, new SMSPeer(EMULATOR_ADDR).getAddress());
+        assertEquals(EMULATOR_ADDR_WITH_PREFIX_AND_EXT, new SMSPeer(EMULATOR_ADDR_WITH_EXT).getAddress());
+        assertEquals(EMULATOR_ADDR_WITH_PREFIX_AND_EXT, new SMSPeer(EMULATOR_ADDR_WITH_PREFIX_AND_EXT).getAddress());
     }
 
-    @Test(expected = InvalidTelephoneNumberException.class)
-    public void constructor_shortTelephoneNumber_trowsException() {
-        new SMSPeer(TOO_SHORT_TELEPHONE_NUMBER);
+    @Test(expected = IllegalArgumentException.class)
+    public void emptyPeer_isNotCreated() {
+        new SMSPeer(EMPTY_ADDR);
+        fail();
```

Comment:
I recently discovered that fails when the tests expects an exception are redundant. If the exception which you are expecting is not throws the test is automatically failed.

Reply:
This comes with Junit4. Fail() was meant to use with this code:
`try{`
  `// do stuff...`
 ` fail("Exception not thrown");`
`}catch(Exception e){`
 ` assertTrue(e.hasSomeFlag());`
`}`
in particular, when the exception needs to be inspected.

Reply:
Aah nice, didn't know that. I'll fix it right now